### ISSUE TITLE
Test failure: Cannot create SSL connection vs Cannot resolve host name

### ISSUE THREAD

**Issue by eserte** [kan/p5-web-service-mailgun#3]
On some of my systems the test suite fails like this:
```
#   Failed test 'error message'
#   at t/03_error.t line 22.
#                   'Cannot create SSL connection: hostname verification failed at /usr/home/cpansand/.cpan/build/2021080206/WebService-Mailgun-0.11-2/blib/lib/WebService/Mailgun.pm line 113.
# '
#     doesn't match '(?^:Cannot resolve host name: )'
# Looks like you failed 1 test of 3.
t/03_error.t ........... 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/3 subtests 
```

**Comment by eserte** [comment]
Well, I don't fully understand it, but it seems that `perl.example.com` actually resolves to something on these systems.

**Comment by kan** [comment]
Thank you for your report.
So, depending on the environment, you may get an SSL error instead of a resolve host name error. Please wait as I will try to fix this issue.

**Comment by eserte** [comment]
Here are some old issues where I observed that *.example.com could be resolved in a surprising way:
* https://rt.cpan.org/Ticket/Display.html?id=107252
* https://rt.cpan.org/Ticket/Display.html?id=120584

**Comment by eserte** [comment]
In the meantime I think I understood the problem. It has nothing to do with SSL, but with wildcard hostnames. This is the same problem as described here: https://github.com/tokuhirom/Furl/issues/128

A potential fix is to add a dot to the non-existing test hostname. In this case the following change seems to help:
```
diff --git a/t/03_error.t b/t/03_error.t
index f100da1..06ccd5f 100755
--- a/t/03_error.t
+++ b/t/03_error.t
@@ -2,7 +2,7 @@ use strict;
 use Test::More 0.98;
 use WebService::Mailgun;

-$WebService::Mailgun::API_BASE = 'api.example.com/v0.1';
+$WebService::Mailgun::API_BASE = 'api.example.com./v0.1';

 my $mailgun = WebService::Mailgun->new(
     api_key => 'key-389807c554fdfe0a7757adf0650f7768',
```

**Comment by kan** [comment]
@eserte 
Thank you for your research and suggestions.
I have released 0.15 which incorporates the fixes you suggested.
You expect to be able to retrieve it from CPAN shortly.
